eduvpn/nm.py

```diff
@@ -138,3 +180,54 @@ def callback(source_object, result):
             logger.debug(source_object.check_connectivity_finish(result))
 
         client.check_connectivity_async(callback=callback)
+
+
+user_dbus_status_callback = None
+vpn_state = ConnectionState.UNKNOWN
+
+
+def register_status_callback(callback):
+    global user_dbus_status_callback
+    user_dbus_status_callback = callback
+    global vpn_state
+    dbus_status_callback(vpn_state, ConnectionStateReason.UNKNOWN)
+
+
+def active_vpn_state() -> ConnectionState:
+    global vpn_state
+    return vpn_state
+
+
+def dbus_status_callback(state_code=None, reason_code=None):
+    global user_dbus_status_callback
+    if user_dbus_status_callback is not None:
+        user_dbus_status_callback(ConnectionState(state_code), ConnectionStateReason(reason_code))
+
+
+def init_dbus_system_bus():
+    """
+    Create a new D-Bus system bus object. We put this here so other modules don't need to import D-Bus
+    """
+    dbus.mainloop.glib.DBusGMainLoop(set_as_default=True)
```

Comment:
i'm not sure if you should start another mainloop. If I remember correctly, starting the glib loop should be enough. 

Also, probably best to not start loops inside functions here, but close to application entry points.

Reply:
It is not starting a new main loop, it's indicating that the Glib main loop should be used:

[Setting up an event loop](https://dbus.freedesktop.org/doc/dbus-python/tutorial.html#setting-up-an-event-loop)